### PR TITLE
MOTH - verify MOT certificate printing

### PR DESCRIPTION
**Pull request by adrianz-kainos**
Please do not merge this yet. We need to enable the proper feature toggle first:
https://jira.i-env.net/browse/BL-9789

### CODE REVIEW

**Comment by adrianz-kainos** [comment]
Please do not merge this yet. We need to enable the proper feature toggle first:
https://jira.i-env.net/browse/BL-9789

File: src/main/resources/features/moth/54 - Print MOT certificate.feature
Comment (ryszardm-kainos) [inline src/main/resources/features/moth/54 - Print MOT certificate.feature]: Scenarios should follow the Given/When/Then pattern - Its hard to understand here what is the path taken and what is the expected result here

**Comment by ibrahimjassat** [comment]
Needs a green run before its merged

File: src/main/resources/features/moth/54 - Print MOT certificate.feature
Comment (ibrahimjassat) [inline src/main/resources/features/moth/54 - Print MOT certificate.feature]: Can include a check if the PDF of the certificate looks correct here

File: src/main/resources/features/moth/54 - Print MOT certificate.feature
Comment (ibrahimjassat) [inline src/main/resources/features/moth/54 - Print MOT certificate.feature]: Would split this into two different tests one for happy path and one for the unhappy path

File: src/main/java/uk/gov/dvsa/mot/fixtures/generic/WebStepDefinitions.java
Comment (karlbeckettbjss) [inline src/main/java/uk/gov/dvsa/mot/fixtures/generic/WebStepDefinitions.java:36]: Changing this step will cause other tests to fail. Is there a reason why it changed?

Comment (adrianz-kainos) [reply src/main/java/uk/gov/dvsa/mot/fixtures/generic/WebStepDefinitions.java:36]: My bad. I was unable to find any usages of this step. I can now see that it's used somewhere else. I will revert this change.

File: src/main/resources/features/moth/54 - Print MOT certificate.feature
Comment (adrianz-kainos) [reply src/main/resources/features/moth/54 - Print MOT certificate.feature]: Checks added

File: src/main/java/uk/gov/dvsa/mot/fixtures/generic/WebStepDefinitions.java
Comment (adrianz-kainos) [reply src/main/java/uk/gov/dvsa/mot/fixtures/generic/WebStepDefinitions.java:36]: I created a new step definition, the one mentioned above should remain intact.

File: src/main/resources/features/moth/54 - Print MOT certificate.feature
Comment (adrianz-kainos) [reply src/main/resources/features/moth/54 - Print MOT certificate.feature]: Done

File: src/main/resources/features/moth/54 - Print MOT certificate.feature
Comment (adrianz-kainos) [reply src/main/resources/features/moth/54 - Print MOT certificate.feature]: Done

**Comment by mrogowska** [comment]
Tests under @mothprint tag passed: https://jenkins2.motdev.org.uk/job/MOTH/job/MOTH_selenium/534/

File: src/main/resources/features/moth/55 - Print MOT certificate successfully.feature
Comment (ibrahimjassat) [inline src/main/resources/features/moth/55 - Print MOT certificate successfully.feature]: Feature number should be changed from 54 to 55 to match the file

**Review by ibrahimjassat — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/dvsa/mot/fixtures/generic/WebStepDefinitions.java
Comment (karlbeckettbjss) [inline src/main/java/uk/gov/dvsa/mot/fixtures/generic/WebStepDefinitions.java:36]: Please add this new step definition to the docs

mot-automated-testsuite/docs/step-definitions.md

File: src/main/java/uk/gov/dvsa/mot/fixtures/mot/DocumentStepDefinitions.java
Comment (karlbeckettbjss) [inline src/main/java/uk/gov/dvsa/mot/fixtures/mot/DocumentStepDefinitions.java:38]: Please add this new step definition to the docs

mot-automated-testsuite/docs/step-definitions.md

**Review by karlbeckettbjss — CHANGES_REQUESTED** [review]

File: src/main/resources/features/moth/55 - Print MOT certificate successfully.feature
Comment (mrogowska) [reply src/main/resources/features/moth/55 - Print MOT certificate successfully.feature]: Number changed.

File: src/main/resources/features/moth/54 - Print MOT certificate with failures.feature
Comment (karlbeckettbjss) [inline src/main/resources/features/moth/54 - Print MOT certificate with failures.feature]: The description for the feature and scenario doesn't match what the tests are doing, can this be changed.

File: src/main/java/uk/gov/dvsa/mot/fixtures/generic/WebStepDefinitions.java
Comment (mrogowska) [reply src/main/java/uk/gov/dvsa/mot/fixtures/generic/WebStepDefinitions.java:36]: Added.

File: src/main/java/uk/gov/dvsa/mot/fixtures/mot/DocumentStepDefinitions.java
Comment (mrogowska) [reply src/main/java/uk/gov/dvsa/mot/fixtures/mot/DocumentStepDefinitions.java:38]: Added.

File: src/main/resources/features/moth/54 - Print MOT certificate with failures.feature
Comment (mrogowska) [reply src/main/resources/features/moth/54 - Print MOT certificate with failures.feature]: The feature description and scenario changed to match what is happening in the tests.

**Review by karlbeckettbjss — APPROVED** [review]

**Review by ibrahimjassat — APPROVED** [review]

**Pull request merged**
